### PR TITLE
Conn 1390:Fix for not allowing to add duplicate domain

### PR DESCRIPTION
**Pull request by sailajaa**
1. Added logic for not to allow adding / updating duplicate domains.Updated related DirectDomainBean.java and other related DAO's and Impl files.
2. Created CreateDomainException class to Instantiate new create domain exception.
3. Updated domaincreatedialog.xhtml, domaineditdialog.xhtml to fix dialogue window refresh issue.
4. Updated configdb.sql,nhincdb.sql and nhincdb_oracle.sql to add unique constraint on domain name in domain table.

### CODE REVIEW

**Comment by nareshcgi** [comment]
Can you add description in the pull request?

**Comment by nareshcgi** [comment]
Can you update the script for oracle?

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/direct/DirectDomainBean.java [hunk 1]
```diff
@@ -162,9 +170,16 @@ public void showEdit() {
...
         UpdateDomain updateDomain = new UpdateDomain();
         updateDomain.setDomain(selectedDomain);
-        directService.updateDomain(updateDomain);
-        selectedDomain = null;
-        refreshDomains();
+        try {
+            directService.updateDomain(updateDomain);
+            selectedDomain = null;
+            refreshDomains();
+        } catch (CreateDomainException domainException) {
```
Comment (christophermay07) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/direct/DirectDomainBean.java]: If we're going to use this Exception for both Create and Modify, should we rename to something more appropriate like "DomainDaoException" or "DomainTxException"?

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/direct/DirectServiceImpl.java [hunk 2]
```diff
@@ -71,26 +72,28 @@
...
+     * @throws CreateDomainException
      */
     @Override
-    public void updateDomain(UpdateDomain domain) {
+    public void updateDomain(UpdateDomain domain) throws CreateDomainException {
         try {
             directProxy.updateDomain(domain);
         } catch (Exception ex) {
-            LOG.error("Unable to update domain: " + domain.getDomain().getDomainName(), ex);
+            throw new CreateDomainException("Duplicate Domain: " + domain.getDomain().getDomainName(), ex);
```
Comment (christophermay07) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/direct/DirectServiceImpl.java]: Is this the only reason update can fail?  If we run into datasource issues, connection problems, constraint violations (excluding name uniqueness), we will see "Duplicate Domain" as the reason in the logs.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/direct/DirectServiceImpl.java [hunk 3]
```diff
@@ -71,26 +72,28 @@
...
+     * @throws CreateDomainException
      */
     @Override
-    public void addDomain(AddDomain domain) {
+    public void addDomain(AddDomain domain) throws CreateDomainException {
         try {
             directProxy.addDomain(domain);
         } catch (Exception ex) {
-            LOG.error("Unable to add new domain: " + domain.getDomain().getDomainName(), ex);
+            throw new CreateDomainException("Duplicate Domain: " + domain.getDomain().getDomainName(), ex);
```
Comment (christophermay07) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/direct/DirectServiceImpl.java]: Same comment as above.

**Comment by christophermay07** [comment]
I've reviewed this PR, endorsing for merge without further comment.

**Comment by nareshcgi** [comment]
Reviewed the PR and looks good.

**Pull request merged**